gcs-proxy.go

```diff
@@ -85,6 +86,9 @@ func (h *GetReqHeader) Requestheaders(f *proxy.Flow) {
 
 func (c *EncryptGcsPayload) Request(f *proxy.Flow) {
 
+	id := uuid.New()
+	f.Request.Header.Add("x-request-id", id.String())
```

Comment:
why is uuid needed? for custom headers, I'd use x-gcsp prefix as Google uses x-goog.  gcsp is short for gcs-proxy.

Reply:
The idea behind adding this header was to uniquely identify the request in logs and metrics if needed during the performance testing. I took reference from here - https://cloud.google.com/healthcare-api/docs/audit-log-http-headers#configure_custom_http_headers
which recommends adding x-request-id custom headers with unique id (using uuid).

Reply:
Please add a comment into the code.

Reply:
@pawanphalak the flow object already has a unique id.  I'd imagine the metrics instrumentation logic can probably use flow ID instead of creating a custom header.